bridge-sdk/connectors/omni-connector/src/omni_connector.rs

```diff
@@ -1,828 +1,680 @@
-        );
+    pub async fn evm_fin_transfer(
+        &self,
+        chain_kind: ChainKind,
+        event: Nep141LockerEvent,
+    ) -> Result<TxHash> {
+        let evm_bridge_client = self.evm_bridge_client(chain_kind)?;
+        evm_bridge_client.fin_transfer(event).await
+    }
 
-        let bridge_token = &self.bridge_token(erc20_address)?;
+    pub async fn evm_fin_transfer_with_tx_hash(
+        &self,
+        chain_kind: ChainKind,
+        near_tx_hash: CryptoHash,
+    ) -> Result<TxHash> {
+        let evm_bridge_client = self.evm_bridge_client(chain_kind)?;
+        let near_bridge_client = self.near_bridge_client()?;
 
-        let signer = self.eth_signer()?;
-        let bridge_token_factory_address = self.bridge_token_factory_address()?;
-        let allowance = bridge_token
-            .allowance(signer.address(), bridge_token_factory_address)
-            .call()
+        let transfer_log = near_bridge_client
+            .extract_transfer_log(near_tx_hash, None, "SignTransferEvent")
             .await?;
 
-        let amount256: ethers::types::U256 = amount.into();
-        if allowance < amount256 {
-            bridge_token
-                .approve(bridge_token_factory_address, amount256 - allowance)
-                .send()
-                .await?
-                .await
-                .map_err(ContractError::from)?;
-
-            tracing::debug!("Approved tokens for spending");
-        }
-
-        let withdraw_call = factory.init_transfer(
-            erc20_address,
-            amount,
-            fee.fee.into(),
-            fee.native_fee.into(),
-            receiver,
-            String::new(),
-        );
-        let tx = withdraw_call.send().await?;
-
-        tracing::info!(
-            tx_hash = format!("{:?}", tx.tx_hash()),
-            "Sent transfer transaction"
-        );
-
-        Ok(tx.tx_hash())
+        evm_bridge_client
+            .fin_transfer(
+                serde_json::from_str(&transfer_log).map_err(|_| BridgeSdkError::UnknownError)?,
+            )
+            .await
     }
 
-    /// Withdraws NEP-141 tokens from the token locker. Requires a proof from the burn transaction
-    #[tracing::instrument(skip_all, name = "NEAR FIN TRANSFER")]
-    pub async fn near_fin_transfer(&self, args: FinTransferArgs) -> Result<CryptoHash> {
-        let near_endpoint = self.near_endpoint()?;
+    pub async fn solana_initialize(&self, program_keypair: Keypair) -> Result<Signature> {
+        // Derived based on near bridge account id and derivation path (bridge-1)
+        const DERIVED_NEAR_BRIDGE_ADDRESS: [u8; 64] = [
+            19, 55, 243, 130, 164, 28, 152, 3, 170, 254, 187, 182, 135, 17, 208, 98, 216, 182, 238,
+            146, 2, 127, 83, 201, 149, 246, 138, 221, 29, 111, 186, 167, 150, 196, 102, 219, 89,
+            69, 115, 114, 185, 116, 6, 233, 154, 114, 222, 142, 167, 206, 157, 39, 177, 221, 224,
+            86, 146, 61, 226, 206, 55, 2, 119, 12,
+        ];
```

Comment:
It would be great to create a function to calculate this. This can be done in a separate PR, as near_bridge_account might be different for testing purposes.

Reply:
Yeah, I asked @kiseln about this already. We will do it later https://github.com/Near-One/bridge-sdk-rs/issues/56